src/client/components/pages/Schedule/ScheduleView.tsx

```diff
@@ -95,19 +109,58 @@ const ScheduleView: FunctionComponent<ScheduleViewProps> = ({
                 / minuteResolution
                 // Add one to account for the header row
               ) + 1;
-              const resolvedDuration = Math.round(duration / minuteResolution);
+              const resolvedDuration = Math.round(
+                duration / minuteResolution
+              );
               return [...blocks, (
                 <SessionBlock
                   key={sessionId}
                   prefix={coursePrefix}
                   startRow={resolvedStartRow}
                   duration={resolvedDuration}
                 >
-                  {courses.map(({ id: courseId, courseNumber }) => (
-                    <CourseListing key={courseId}>
-                      {courseNumber}
-                    </CourseListing>
-                  ))}
+                  {courses.map(({
+                    id: courseId,
+                    courseNumber,
+                    room,
+                  }) => {
+                    const catalogNumber = `${coursePrefix} ${courseNumber}`;
+                    const displayStartTime = PGTime.toDisplay(
+                      `${startHour}:${startMinute.toString().padStart(2, '0')}`
+                    );
+                    const displayEndTime = PGTime.toDisplay(
+                      `${endHour}:${endMinute.toString().padStart(2, '0')}`
+                    );
+                    return (
+                      <CourseListing key={courseId}>
+                        <CourseListingButton
+                          aria-hidden
+                          onClick={(event) => {
+                            event.stopPropagation();
+                            setCurrentPopover((current) => (
+                              current === courseId ? null : courseId
+                            ));
+                          }}
+                        >
+                          {courseNumber}
+                        </CourseListingButton>
+                        <Popover
+                          aria-hidden
+                          xOffset="0.5rem"
+                          yOffset="1rem"
+                          title={catalogNumber}
+                          isVisible={currentPopover === courseId}
+                        >
```

Comment:
Should we calculate where the user has clicked so that the `Popover` can show up on the left side for the items on the right of the schedule?

Reply:
Ok got it, thanks for clarifying!